### DISCUSSION TITLE
Trigger actions workflow only when push was made to the currently-default branch

### DISCUSSION THREAD

**Discussion by williewillus** [community/community#64921] [Actions] [answered]
### Select Topic Area

Question

## What I Want to Do
Trigger a Github Action, only if it was triggered by a push event to a branch which is the **current default** branch of the repository.

## Why
Part of my repository has a website that is built and deployed on push. The repository is periodically branched into new versioned branches, so there is no single `main` or `master` branch. The currently most-supported version is the default branch, and it switches every year or so.

However, sometimes older branches need to be hotfixed. When those branches push, the website build job then attempts to build and deploy the old website on that branch, which is not what I want.

This means in order to no clobber my website, I have to go back and disable the web push job for all the old branches, and also have to keep updating the `on: push: <branch>` job to have the name of my current default branch.

Actions supports `on: push: <branch>`, but this requires changing the branch name every time I update my default branch.

## Question
How do I do it?
Thanks in advance.

**Comment by spenserblack** [comment]
:thinking: That's an interesting problem. So, for example, you might have a `v3` branch that is the default branch, but you need to commit to `v2`?

Sorry if this isn't the answer you'd like :laughing:
But TBH I'd recommend using `main` or `master` for the latest supported version, and using versioned branches for *older,* non-default versions. You could perhaps use `latest` instead of `main` to make that more clear. That's just what I think would be easiest long-term IMO.

But you can also skip the CI with with special commit messages. [Here's some documentation on that](https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs). Or, *before* you create your new branch, add a commit to change the target branch for the action so it won't run on other branches.

You might want to even create a "new branch" GitHub action that runs on workflow dispatch, to create a new branch and disable the old workflows.

**Comment by williewillus** [comment, marked as answer]
Using main isn't an option, as I mentioned.

What I ended up doing which works, for posterity, is to fetch the default_branch of the repo using an API call, then comparing that with `github.ref`, and running the job only when the two match (modulo `refs/head/` prefix)

**Reply by Zeratoxx** [reply]
So you have a workflow that triggers at every push in any branch and its first step is to look for the default branch and compare it to the `ref`?

**Comment by Zeratoxx** [comment]
Why can GitLab support this behavior but GitHub cannot?  
This feature is really important in my opinion.

**Comment by brazarb** [comment]
I agree this should be implemented. The starter workflows already support the $default-branch syntax, maybe we can do something similar?

```
name: Octo Organization CI

on:
  push:
    branches: [ $default-branch ]
  pull_request:
    branches: [ $default-branch ]

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v4

      - name: Run a one-line script
        run: echo Hello from Octo Organization
```

https://docs.github.com/en/actions/using-workflows/creating-starter-workflows-for-your-organization
